.vscode/launch.json

```diff
@@ -7,60 +7,59 @@
             "request": "launch",
             "runtimeExecutable": "${execPath}",
             "args": [
-                "--extensionDevelopmentPath=${workspaceRoot}"
+                "${workspaceFolder}/test/integrationTests/testAssets/singleCsproj",
+                "--extensionDevelopmentPath=${workspaceFolder}"
             ],
             "stopOnEntry": false,
             "sourceMaps": true,
             "outFiles": [
-                "${workspaceRoot}/out/src/**/*.js"
+                "${workspaceFolder}/out/src/**/*.js"
             ]
         },
         {
-            "name": "Launch Unit Tests",
+            "name": "Launch Extension with singleCsproj",
             "type": "extensionHost",
             "request": "launch",
             "runtimeExecutable": "${execPath}",
             "args": [
-                "--extensionDevelopmentPath=${workspaceRoot}",
-                "--extensionTestsPath=${workspaceRoot}/out/test/unitTests"
+                "${workspaceFolder}/test/integrationTests/testAssets/singleCsproj",
+                "--extensionDevelopmentPath=${workspaceFolder}"
             ],
             "stopOnEntry": false,
             "sourceMaps": true,
             "outFiles": [
-                "${workspaceRoot}/out/test/**/*.js"
+                "${workspaceFolder}/out/test/**/*.js"
```

Comment:
>{workspaceFolder} [](start = 18, length = 17)

Can you explain this? It looks like VSCode deprecated workspaceroot for workspacefolder?

Reply:
That's right. `workspaceRoot` was deprecated when multi-root workspaces were added to VS Code.